Rework rendering of experience indicator

ihhub opened the pull request:
- all values are going to be rendered strictly within the given black area.
- if an experience string is too long "M" (million) notation is going to be used

relates to #9433

Branikolog commented:
Hi, @ihhub !

"1" is missing its most left pixel for this case:
![image](https://github.com/user-attachments/assets/624ca8a1-a56c-42fd-befc-96bb1818ee35)

Also, after 199 it seems for some shorter numbers the full version is rendered, and for wider numbers - the contracted version. It looks not really consistent.

https://github.com/user-attachments/assets/bdc47fbe-dca9-46fc-827b-b9c5dae50ba4
Since the proposed fix does not makes the field wider I suggest showing the contracted mana representation in a consistent way, after a certain values.
What if we just render the contracted version for all cases above 199/990 and 99/1000? Usually players are not really bothered about the maximum available points, when they have so much.

ihhub commented:
Hi @Branikolog , could you please give me your map or save file for tests?

Branikolog commented:
Hi, @ihhub
Here's the map with cut "1" symbol and two heroes 201 and 202 mana:
[Mana+exp.zip](https://github.com/user-attachments/files/18587465/Mana%2Bexp.zip)

Branikolog commented:
@Mr-Bajs , hi!

> The right click dialog on the EXP bar shows a detailed info on the EXP and next level. The spell points does not however.
> 
> If the player would want more detailed info in the exact number of spell points, its accessible when right clicking the heros portrait. But shouln't the exact and and total spell point also be showed when right clicking in the spell points book/bar, below the explainatory text?

Isn't it working in the game as you've described already?
![image](https://github.com/user-attachments/assets/a41a17bf-cf31-4c84-a74b-bf02bc64892a)

ihhub commented:
Hi @Branikolog , to move forward I reverted changes related to spell points as they ignited a discussion. For now, this pull request fixes only experience string. Also, I've fixed horizontal location of the experience string. Please take a look.

ihhub commented:
@Branikolog and @Districh-ru , I am merging this pull request as we are close to the release. Please use the master branch for further testing.
